mountpoint-s3-fs/src/superblock/readdir.rs

```diff
@@ -204,9 +212,30 @@ impl ReaddirHandle {
 /// should be done lazily by the consumer of the entry.
 #[derive(Debug, Clone)]
 enum ReaddirEntry {
-    RemotePrefix { name: String },
-    RemoteObject { name: String, object_info: ObjectInfo },
-    LocalInode { lookup: LookedUp },
+    RemotePrefix {
+        name: String,
+    },
+    RemoteObject {
+        /// Last component of the S3 Key.
+        name: String,
+        /// S3 Key for this object.
+        full_key: String,
```

Comment:
nit: is the key this needed?

Reply:
it's used for a couple of warnings:
```
warn!("{} has an invalid name and will be unavailable", next.description());
warn!(
    "{} is omitted because another {} exist with the same name",
    entry.description(),
    last_entry.description(),
);
```

we don't log the `full_key` for `RemotePrefix` (only `name`) so may be we can ditch it:
```
Self::RemotePrefix { name } => {
    format!("directory '{name}'")
}
Self::RemoteObject { name, full_key, .. } => {
    format!("file '{}' (full key {:?})", name, full_key)
}
```

but I think parent inode is not available at the point of logging, so there's no simple way to restore `full_key` from `name`

Reply:
No worries, I had missed that.